osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs

```diff
@@ -229,5 +232,33 @@ public ProfileCoverBackground()
                 Masking = true;
             }
         }
+
+        private partial class UpdateableCountryText : OsuHoverContainer
+        {
+
+            public FontUsage Font = default;
+
+            [Resolved]
+            private RankingsOverlay? rankingsOverlay { get; set; }
+
+            public UpdateableCountryText()
+            {
+                AutoSizeAxes = Axes.Both;
+            }
+
+            public void load(CountryCode countryCode)
+            {
+                Action = () =>
+                {
+                    rankingsOverlay?.ShowCountry(countryCode);
+                };
+
+                Child = new OsuSpriteText
+                {
+                    Font = Font,
+                    Text = countryCode.GetDescription(),
+                };
+            }
+        }
```

Comment:
You don't need to create a new class for this. You can just create a new instance of `OsuHoverContainer` and use the object initializer syntax to initialize `Action`, `AutoSizeAxes` and `Child`

Reply:
Thanks, i understand the project stucture much better right now. But i have a question about the code inspection. The JetBrains Inspect Code creates a huge report file and i'm pretty sure it shouldn't do that.
![image](https://user-images.githubusercontent.com/46729135/235044014-fbd37589-b612-4ead-8c4b-2c1823b721d5.png)

Reply:
This is to be studied with ReSharper, however it is paid AFAIK. But you can take a glance at the file to see what the issues are (shouldn't be difficult to figure out what all of this means), or you can search online for softwares that can parse this.